### PR TITLE
feat: DaskLazyFrame `join` and `sort`, DaskNamespace `all`

### PR DESCRIPTION
**Pull request by FBruzzesi**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue #637 

## Checklist

- [x] Code follows style guide (ruff)
- [x] Tests added 
- [ ] Documented the changes

## If you have comments or can explain your changes, please do so below.

It ended up being significantly larger than what it was intended.

Namely, wanted to implement `join` method, yet in the test we use `sort` method and `nw.all` function. 
I can split it up in 2 or 3 if needed

### CODE REVIEW

File: narwhals/_dask/dataframe.py [hunk 1]
```diff
@@ -123,3 +128,113 @@ def with_row_index(self: Self, name: str) -> Self:
 
     def rename(self: Self, mapping: dict[str, str]) -> Self:
         return self._from_native_dataframe(self._native_dataframe.rename(columns=mapping))
+
+    def sort(
```
Comment (FBruzzesi) [inline narwhals/_dask/dataframe.py:132]: Same as pandas-like implementation

File: narwhals/_dask/dataframe.py [hunk 2]
```diff
@@ -123,3 +128,113 @@ def with_row_index(self: Self, name: str) -> Self:
...
+    ) -> Self:
+        flat_keys = flatten([*flatten([by]), *more_by])
+        df = self._native_dataframe
+        if isinstance(descending, bool):
+            ascending: bool | list[bool] = not descending
+        else:
+            ascending = [not d for d in descending]
+        return self._from_native_dataframe(df.sort_values(flat_keys, ascending=ascending))
+
+    def join(
```
Comment (FBruzzesi) [inline narwhals/_dask/dataframe.py:146]: Also same as pandas, except that `how="cross" ` is never supported in Dask

File: narwhals/_dask/namespace.py [hunk 3]
```diff
@@ -50,6 +50,18 @@ def col(self, *column_names: str) -> DaskExpr:
             backend_version=self._backend_version,
         )
 
+    def all(self) -> DaskExpr:
+        return DaskExpr(
+            lambda df: [
+                df._native_dataframe.loc[:, column_name] for column_name in df.columns
+            ],
```
Comment (FBruzzesi) [inline narwhals/_dask/namespace.py:57]: I feel somehow a bit uncomfy returning a native object, but since we don't have a DaskSeries I think that's ok?!

Comment (MarcoGorelli) [reply narwhals/_dask/namespace.py:57]: yup!

**Review by MarcoGorelli — APPROVED** [review]
coooool thanks @FBruzzesi !

**Pull request merged**